underscore.js

```diff
@@ -1213,6 +1213,11 @@
     // Assume equality for cyclic structures. The algorithm for detecting cyclic
     // structures is adapted from ES 5.1 section 15.12.3, abstract operation `JO`.
 
+    // Compare array lengths to determine if a deep comparison is necessary.
+    if (areArrays && a.length !== b.length) {
+      return false;
+    }
+    
```

Comment:
Well, I need to re-read the code (am on mobile) but there are "areArrays" check below-I'm afraid your code would make `{length: 3, a: 1}` count as an array... But it might already be the case